core/src/layers/dtrace.rs

```diff
@@ -0,0 +1,440 @@
+        let c_path = CString::new(path).unwrap();
+        probe_lazy!(opendal, blocking_create_dir_start, c_path.as_ptr());
+        let result = self.inner.blocking_create_dir(path, args);
+        probe_lazy!(opendal, blocking_create_dir_end, c_path.as_ptr());
+        result
+    }
+
+    fn blocking_read(&self, path: &str, args: OpRead) -> Result<(RpRead, Self::BlockingReader)> {
+        let c_path = CString::new(path).unwrap();
+        probe_lazy!(opendal, fetch_blocking_reader_start, c_path.as_ptr());
+        let result = self
+            .inner
+            .blocking_read(path, args)
+            .map(|(rp, r)| (rp, DtraceLayerWarpper::new(r, &path.to_string())));
+        probe_lazy!(opendal, fetch_blocking_reader_end, c_path.as_ptr());
+        result
+    }
+
+    fn blocking_write(&self, path: &str, args: OpWrite) -> Result<(RpWrite, Self::BlockingWriter)> {
+        let c_path = CString::new(path).unwrap();
+        probe_lazy!(opendal, fetch_blocking_writer_start, c_path.as_ptr());
+        let result = self
+            .inner
+            .blocking_write(path, args)
+            .map(|(rp, r)| (rp, DtraceLayerWarpper::new(r, &path.to_string())));
+        probe_lazy!(opendal, fetch_blocking_writer_end, c_path.as_ptr());
+        result
+    }
+
+    fn blocking_stat(&self, path: &str, args: OpStat) -> Result<RpStat> {
+        let c_path = CString::new(path).unwrap();
+        probe_lazy!(opendal, blocking_stat_start, c_path.as_ptr());
+        let result = self.inner.blocking_stat(path, args);
+        probe_lazy!(opendal, blocking_stat_end, c_path.as_ptr());
+        result
+    }
+
+    fn blocking_delete(&self, path: &str, args: OpDelete) -> Result<RpDelete> {
+        let c_path = CString::new(path).unwrap();
+        probe_lazy!(opendal, blocking_delete_start, c_path.as_ptr());
+        let result = self.inner.blocking_delete(path, args);
+        probe_lazy!(opendal, blocking_delete_end, c_path.as_ptr());
+        result
+    }
+
+    fn blocking_list(&self, path: &str, args: OpList) -> Result<(RpList, Self::BlockingLister)> {
+        let c_path = CString::new(path).unwrap();
+        probe_lazy!(opendal, blocking_list_start, c_path.as_ptr());
+        let result = self.inner.blocking_list(path, args);
+        probe_lazy!(opendal, blocking_list_end, c_path.as_ptr());
+        result
+    }
+}
+
+pub struct DtraceLayerWarpper<R> {
+    inner: R,
+    path: String,
+}
+
+impl<R> DtraceLayerWarpper<R> {
+    pub fn new(inner: R, path: &String) -> Self {
+        Self {
+            inner,
+            path: path.to_string(),
+        }
+    }
+}
+
+impl<R: oio::Read> oio::Read for DtraceLayerWarpper<R> {
+    fn poll_read(&mut self, cx: &mut Context<'_>, buf: &mut [u8]) -> Poll<Result<usize>> {
+        let c_path = CString::new(self.path.clone()).unwrap();
+        probe_lazy!(opendal, poll_read_start, c_path.as_ptr());
+        self.inner.poll_read(cx, buf).map(|res| match res {
+            Ok(bytes) => {
```

Comment:
It's better to use `n` to make it more clear that it's the length of read bytes instead of content.